### DISCUSSION TITLE
systemctl --user enable onedrive does not find service on Ubuntu 24.10

### DISCUSSION THREAD

**Discussion by arvedjo** [abraunegg/onedrive#2975] [Q&A] [answered]
When I use the installation as described in the man pages for Ubuntu 24.10 the service file seems not to be installed. 
I cannot find the onedrive service with `systemctl list-unit-files` and running `systemctl --user enable onedrive`
Returns `Failed to enable unit: Unit onedrive.service does not exist`. Any idea how to fix it?
Distributor ID:	Ubuntu
Description:	Ubuntu 24.10
Release:	24.10
Codename:	oracular

**Comment by abraunegg** [comment]
@arvedjo 
> When I use the installation as described in the man pages for Ubuntu 24.10
> ....
> Any idea how to fix it?

This is the *wrong* way to install the client.

#### Actions for you
* Read all documentation on this GitHub site
* Install the client the *correct* way

**Comment by arvedjo** [comment]
Hi @abraunegg thanks for the quick reply. To clarify I read the documentation from this GitHub site und used the Suse mirror.

**Comment by abraunegg** [comment]
@arvedjo 

> Hi @abraunegg thanks for the quick reply. To clarify I read the documentation from this GitHub site und used the Suse mirror.

I have installed Ubuntu 24.10 and can confirm your issue:

```
alex@ubuntu-2410:~$ onedrive --version
onedrive v2.5.2-1+np1+1.1
alex@ubuntu-2410:~$ systemctl list-unit-files | grep onedrive
alex@ubuntu-2410:~$ systemctl --user enable onedrive
Failed to enable unit: Unit onedrive.service does not exist
alex@ubuntu-2410:~$ 
alex@ubuntu-2410:~$ 
alex@ubuntu-2410:~$ 
```
Check `/usr/lib/systemd/user` directory:
```
root@ubuntu-2410:/usr/lib/systemd/user# ls -la onedrive*
ls: cannot access 'onedrive*': No such file or directory
root@ubuntu-2410:/usr/lib/systemd/user# 
```


On Ubuntu 22.04 | 24.04 the files are installed in `/usr/lib/systemd/user` as per:

```
alex@ubuntu-24-04:/usr/lib/systemd/user$ ls onedrive.service 
onedrive.service
alex@ubuntu-24-04:/usr/lib/systemd/user$ 
```

So on Ubuntu 24.10 they are not being installed. Additionally, they are not being installed *anywhere* as per:
```
root@ubuntu-2410:/# find . | grep 'onedrive.service' 
find: ‘./run/user/120/doc’: Permission denied
find: ‘./run/user/1000/doc’: Permission denied
root@ubuntu-2410:/# find . | grep 'onedrive@.service' 
find: ‘./run/user/120/doc’: Permission denied
find: ‘./run/user/1000/doc’: Permission denied
root@ubuntu-2410:/# 
```

So this is a *packaging* issue - not a bug with the application.

@norbusan  - please can you look into this issue as to why the systemd files are not being installed on Ubuntu 24.10 in the correct location when creating the v2.5.3 packages

@sbadia - potentially can you provide some insights here as well.

**Reply by norbusan** [reply]
Very strange, the package is the same ... Hmm, had something changed in his systemd unit files are installed?

Someone with a 24.10, can you please check one of the postinst scripts in /var/lib/dpkg/info/ how unit files are installed? 

I'll check, too, later tonight or tomorrow.

**Reply by abraunegg** [reply]
@norbusan 
I have build a Ubuntu 24.10 system .. let me know what you need - from the list file:
```
root@ubuntu-2410:/var/lib/dpkg/info# cat onedrive.list 
/.
/etc
/etc/default
/etc/init.d
/etc/logrotate.d
/etc/logrotate.d/onedrive
/usr
/usr/bin
/usr/bin/onedrive
/usr/share
/usr/share/bash-completion
/usr/share/bash-completion/completions
/usr/share/bash-completion/completions/onedrive
/usr/share/doc
/usr/share/doc/onedrive
/usr/share/doc/onedrive/LICENSE.gz
/usr/share/doc/onedrive/NEWS.Debian.gz
/usr/share/doc/onedrive/advanced-usage.md.gz
/usr/share/doc/onedrive/application-config-options.md.gz
/usr/share/doc/onedrive/application-security.md.gz
/usr/share/doc/onedrive/business-shared-items.md.gz
/usr/share/doc/onedrive/changelog.Debian.gz
/usr/share/doc/onedrive/changelog.md.gz
/usr/share/doc/onedrive/client-architecture.md.gz
/usr/share/doc/onedrive/config
/usr/share/doc/onedrive/contributing.md.gz
/usr/share/doc/onedrive/copyright
/usr/share/doc/onedrive/docker.md.gz
/usr/share/doc/onedrive/install.md.gz
/usr/share/doc/onedrive/known-issues.md.gz
/usr/share/doc/onedrive/national-cloud-deployments.md.gz
/usr/share/doc/onedrive/podman.md.gz
/usr/share/doc/onedrive/privacy-policy.md.gz
/usr/share/doc/onedrive/readme.md.gz
/usr/share/doc/onedrive/sharepoint-libraries.md.gz
/usr/share/doc/onedrive/terms-of-service.md
/usr/share/doc/onedrive/ubuntu-package-install.md.gz
/usr/share/doc/onedrive/usage.md.gz
/usr/share/doc/onedrive/webhooks.md.gz
/usr/share/fish
/usr/share/fish/vendor_completions.d
/usr/share/fish/vendor_completions.d/onedrive.fish
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/onedrive.1.gz
/usr/share/zsh
/usr/share/zsh/vendor-completions
/usr/share/zsh/vendor-completions/_onedrive
root@ubuntu-2410:/var/lib/dpkg/info# 
```

**Comment by norbusan** [comment]
To be more specific, I am using in the `debian/rules` files the following stanza:

```
override_dh_installsystemduser:
        dh_installsystemduser --no-enable
```

That has installed the systemd unit files since .... since .... since the introduction of systemd in Debian/Ubuntu, so I am surprised.

**Reply by abraunegg** [reply]
```
root@ubuntu-2410:/var/lib/dpkg/info# cat onedrive.postinst 
#!/bin/sh
set -e
# Automatically added by dh_installdeb/13.18ubuntu2
dpkg-maintscript-helper rm_conffile /etc/onedrive.conf 2.2.1-1\~ -- "$@"
# End automatically added section
root@ubuntu-2410:/var/lib/dpkg/info# 
```

**Reply by abraunegg** [reply]
> To be more specific, I am using in the `debian/rules` files the following stanza:
> 
> ```
> override_dh_installsystemduser:
>         dh_installsystemduser --no-enable
> ```
> 
> That has installed the systemd unit files since .... since .... since the introduction of systemd in Debian/Ubuntu, so I am surprised.

Yep ... so something in Ubuntu 24.10 has changed (what a surprise ...)

**Comment by abraunegg** [comment]
Comparing this to Ubuntu 22.04:
```
alex@ubuntu-22-04:/var/lib/dpkg/info$ cat onedrive.list 
/.
/etc
/etc/default
/etc/init.d
/etc/logrotate.d
/etc/logrotate.d/onedrive
/lib
/lib/systemd
/lib/systemd/system
/lib/systemd/system/onedrive@.service
/usr
/usr/bin
/usr/bin/onedrive
/usr/lib
/usr/lib/systemd
/usr/lib/systemd/user
/usr/lib/systemd/user/onedrive.service
/usr/share
/usr/share/bash-completion
/usr/share/bash-completion/completions
/usr/share/bash-completion/completions/onedrive
/usr/share/doc
/usr/share/doc/onedrive
/usr/share/doc/onedrive/LICENSE.gz
/usr/share/doc/onedrive/NEWS.Debian.gz
/usr/share/doc/onedrive/advanced-usage.md.gz
/usr/share/doc/onedrive/application-config-options.md.gz
/usr/share/doc/onedrive/application-security.md.gz
/usr/share/doc/onedrive/business-shared-items.md.gz
/usr/share/doc/onedrive/changelog.Debian.gz
/usr/share/doc/onedrive/changelog.md.gz
/usr/share/doc/onedrive/client-architecture.md.gz
/usr/share/doc/onedrive/config
/usr/share/doc/onedrive/contributing.md.gz
/usr/share/doc/onedrive/copyright
/usr/share/doc/onedrive/docker.md.gz
/usr/share/doc/onedrive/install.md.gz
/usr/share/doc/onedrive/known-issues.md.gz
/usr/share/doc/onedrive/national-cloud-deployments.md.gz
/usr/share/doc/onedrive/podman.md.gz
/usr/share/doc/onedrive/privacy-policy.md.gz
/usr/share/doc/onedrive/readme.md.gz
/usr/share/doc/onedrive/sharepoint-libraries.md.gz
/usr/share/doc/onedrive/terms-of-service.md
/usr/share/doc/onedrive/ubuntu-package-install.md.gz
/usr/share/doc/onedrive/usage.md.gz
/usr/share/doc/onedrive/webhooks.md.gz
/usr/share/fish
/usr/share/fish/vendor_completions.d
/usr/share/fish/vendor_completions.d/onedrive.fish
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/onedrive.1.gz
/usr/share/zsh
/usr/share/zsh/vendor-completions
/usr/share/zsh/vendor-completions/_onedrive
alex@ubuntu-22-04:/var/lib/dpkg/info$ 
```

**Comment by abraunegg** [comment]
Comparing this to Ubuntu 24.04:
```
alex@ubuntu-24-04:/var/lib/dpkg/info$ cat onedrive.list 
/.
/etc
/etc/default
/etc/init.d
/etc/logrotate.d
/etc/logrotate.d/onedrive
/usr
/usr/bin
/usr/bin/onedrive
/usr/lib
/usr/lib/systemd
/usr/lib/systemd/system
/usr/lib/systemd/system/onedrive@.service
/usr/lib/systemd/user
/usr/lib/systemd/user/onedrive.service
/usr/share
/usr/share/bash-completion
/usr/share/bash-completion/completions
/usr/share/bash-completion/completions/onedrive
/usr/share/doc
/usr/share/doc/onedrive
/usr/share/doc/onedrive/LICENSE.gz
/usr/share/doc/onedrive/NEWS.Debian.gz
/usr/share/doc/onedrive/advanced-usage.md.gz
/usr/share/doc/onedrive/application-config-options.md.gz
/usr/share/doc/onedrive/application-security.md.gz
/usr/share/doc/onedrive/business-shared-items.md.gz
/usr/share/doc/onedrive/changelog.Debian.gz
/usr/share/doc/onedrive/changelog.md.gz
/usr/share/doc/onedrive/client-architecture.md.gz
/usr/share/doc/onedrive/config
/usr/share/doc/onedrive/contributing.md.gz
/usr/share/doc/onedrive/copyright
/usr/share/doc/onedrive/docker.md.gz
/usr/share/doc/onedrive/install.md.gz
/usr/share/doc/onedrive/known-issues.md.gz
/usr/share/doc/onedrive/national-cloud-deployments.md.gz
/usr/share/doc/onedrive/podman.md.gz
/usr/share/doc/onedrive/privacy-policy.md.gz
/usr/share/doc/onedrive/readme.md.gz
/usr/share/doc/onedrive/sharepoint-libraries.md.gz
/usr/share/doc/onedrive/terms-of-service.md
/usr/share/doc/onedrive/ubuntu-package-install.md.gz
/usr/share/doc/onedrive/usage.md.gz
/usr/share/doc/onedrive/webhooks.md.gz
/usr/share/fish
/usr/share/fish/vendor_completions.d
/usr/share/fish/vendor_completions.d/onedrive.fish
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/onedrive.1.gz
/usr/share/zsh
/usr/share/zsh/vendor-completions
/usr/share/zsh/vendor-completions/_onedrive
alex@ubuntu-24-04:/var/lib/dpkg/info$ 
```

**Comment by norbusan** [comment]
Chekcingg the build logs, the installation of systemd unit files is missing, and that is guarded by
```
ifeq ($(HAVE_SYSTEMD),yes)
```
I need to check why this is not set on 24.10

**Comment by norbusan** [comment]
The confugure output is COMPLETELY the same???  
@abraunegg did you change something here in configure.ac:
```
AS_IF([test "x$with_systemdsystemunitdir" = "xyes" -o "x$with_systemdsystemunitdir" = "xauto"],
      [ dnl true part, so try to determine with pkg-config
        def_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)
        dnl if we cannot find it via pkg-config, *and* the user explicitly passed it in with,
        dnl we warn, and in all cases we unset (set to no) the respective variable
        AS_IF([test "x$def_systemdsystemunitdir" = "x"],
              [  dnl we couldn't find the default value via pkg-config
                AS_IF([test "x$with_systemdsystemunitdir" = "xyes"],
                      [AC_MSG_ERROR([systemd support requested but pkg-config unable to query systemd package])])
                with_systemdsystemunitdir=no
              ],
              [ dnl pkg-config found the value, use it
                with_systemdsystemunitdir="$def_systemdsystemunitdir"
              ]
        )
      ]
)
```

**Reply by abraunegg** [reply]
> The confugure output is COMPLETELY the same??? @abraunegg did you change something here in configure.ac:
> 
> ```
> AS_IF([test "x$with_systemdsystemunitdir" = "xyes" -o "x$with_systemdsystemunitdir" = "xauto"],
>       [ dnl true part, so try to determine with pkg-config
>         def_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)
>         dnl if we cannot find it via pkg-config, *and* the user explicitly passed it in with,
>         dnl we warn, and in all cases we unset (set to no) the respective variable
>         AS_IF([test "x$def_systemdsystemunitdir" = "x"],
>               [  dnl we couldn't find the default value via pkg-config
>                 AS_IF([test "x$with_systemdsystemunitdir" = "xyes"],
>                       [AC_MSG_ERROR([systemd support requested but pkg-config unable to query systemd package])])
>                 with_systemdsystemunitdir=no
>               ],
>               [ dnl pkg-config found the value, use it
>                 with_systemdsystemunitdir="$def_systemdsystemunitdir"
>               ]
>         )
>       ]
> )
> ```

No changes to configure.ac AFAIK ....

**Comment by norbusan** [comment]
Anyway, I will pass in `--with-systemdsystemunitdir=auto` and see if that fixes it.
WDYT?

**Comment by norbusan** [comment]
New build sent off.

**Comment by norbusan** [comment]
Didn't work either ...

**Reply by abraunegg** [reply]
> Didn't work either ...

Very odd if auto does not work ....

**Reply by norbusan** [reply]
Absolutely cryptic! There is no warning, and systemd manag of 24.10 show the same pkg-config command ...???

**Reply by norbusan** [reply]
I guess I need to pull a docker 2410 image and see what is going on

**Comment by norbusan** [comment]
Yup, 24.10 gives this:
```
root@38726770a750:~# pkg-config systemd --variable=systemdsystemconfdir 
root@38726770a750:~# 
```

**Reply by norbusan** [reply]
Even after installing libsystemd-dev ?

**Comment by norbusan** [comment]
Found it ... we need `systemd-dev` ... not sure this package is available in older distributions ... let me check

**Comment by norbusan** [comment]
Next build sent off

**Comment by norbusan** [comment]
Ok, systemd-dev was introduced in 23.10, and in 24.10 it is not automatically installed anymore.
Now that I have added it as b-d, older builds are broken.
I need to see how I can add it dynamically depending on the version. What a bummer.

**Reply by abraunegg** [reply]
> Ok, systemd-dev was introduced in 23.10, and in 24.10 it is not automatically installed anymore. Now that I have added it as b-d, older builds are broken. I need to see how I can add it dynamically depending on the version. What a bummer.

Thats crap ... but .. thats ~~Crapbuntu~~ Ubuntu for you ...

**Comment by norbusan** [comment, marked as answer]
Fixed, and updated to 2.5.3

**Comment by abraunegg** [comment]
@norbusan 
Confirmed - greatly appreciated:

![image](https://github.com/user-attachments/assets/4cf451cf-f1e5-4031-a42b-1413eaf3c4ad)

@arvedjo 

Please update / upgrade to v2.5.3 and the updated package and retry your command:

```
alex@ubuntu-2410:~$ systemctl --user enable onedrive
Created symlink '/home/alex/.config/systemd/user/default.target.wants/onedrive.service' → '/usr/lib/systemd/user/onedrive.service'.
alex@ubuntu-2410:~$ 

```

**Reply by arvedjo** [reply]
Hi @abraunegg @norbusan ,
Thanks so much for the fix! Everything works just fine. What an impressive speed as well I am truely impressed!

**Reply by norbusan** [reply]
> What an impressive speed as well I am truely impressed!

You were lucky that it was Saturday and my daughter in school for an event, so I had free time at hand, a rarity!
